### PR TITLE
build: add functions.json file for plugin

### PR DESCRIPTION
**Pull request by Mause**

### CODE REVIEW

File: scripts/generate_function_json.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                {
+                    **function,
+                    'parameters': (
+                        function['parameters'].split(',')
+                        if function.get('parameters')
+                        else
+                        []
+                    ),
+                    'category': category,
+                    'result': get_result(function['example']) if function.get('example') else None
```
Comment (carlopi) [inline scripts/generate_function_json.py:51]: Pretty cool the self-check on results!

Comment (Mause) [reply scripts/generate_function_json.py:51]: I actually found some issues due to this, going to submit a pr to the main repo shortly

File: scripts/generate_function_json.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                }
+                for function in json.load(fh)
+            ]
+
+    functions = sorted(functions, key=lambda fn: fn['name'])
+
+    with open('docs/functions.json', 'w') as fh:
```
Comment (carlopi) [inline scripts/generate_function_json.py:58]: One small addition I would consider (even later) is given that we need a binary, we can add a field (either to `docs/functions.json` or some ancillary file) like: "duckdb_version": "some_git_hash". I don't see a strong use right now but for avoiding re-generating if fields is the same or tracking down weird behavioiur, but I guess the extra cost is low.

File: scripts/generate_function_json.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    except CalledProcessError as e:
+        print(example.strip(), e.stderr.decode())
+        return None
+
+
+def main():
+    functions = []
+    source = Path(args.source)
+    assert source.exists()
+    for file in source.glob('src/core_functions/**/*.json'):
```
Comment (carlopi) [inline scripts/generate_function_json.py:38]: The only improvement I could see here is that eventually we might want to go full on with duckfooding, where we use duckdb itself (either from within python or as a bash script) to generate functions.json. 🦆

**Review by carlopi — APPROVED** [review]
Thanks @Mause! Looks great!

File: scripts/generate_function_json.py (hunk 2 above)
Comment (Mause) [reply scripts/generate_function_json.py:58]: Given the way this stuff works is that we don't commit the new file if it hasn't changed, maybe.

But yeah, the fact that the times in this file change on each run is annoying

**Pull request merged**